### PR TITLE
Handle PrimaryKey as str on Python3

### PR DESCRIPTION
**Pull request by cielavenir**
I can see PrimaryKey is converted from unicode to str on Python2 and str to bytes on Python3. The thing is, bytes is not string type on Python3. It cannot be passed to webstackclient etc normally. I need to decode the bytes in commonui etc. I don't think that's correct way.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+- Handle PrimaryKey as str on Python3
```
Comment (cielavenir) [inline CHANGELOG.md:1]: I will add minor after approval, not patch

**Comment by cielavenir** [comment]
pipelineid=373164 Pipeline #589117

**Comment by cielavenir** [comment]
alternatively you can paste this message to `mugpg -d` to see the URL directly

```
-----BEGIN PGP MESSAGE-----

hQIMAwAinXiLUbjzARAAh8wNmWHdtsYhYgat5FmRi9HKKt+UqP1P6lQ+zGHm3Omu
wxobxqedprkZE0137aboJ4o9wSYCPWI6ROznMZGYE8tsYfUrMczEuzAiyaqqiNVE
oSw77N+5X03p65rAsxhnelYvKLUAt6j97CgBGZHXseVKobfgB8J831E47/ggIIQ7
jUeOjJxqzD6oHZBW4tLZ95utWfy393d5rVdxvjBkqpK9bj4tfhsN0WoSYkyGlPnR
Uqakso/hIW/R7kthXd0j8sSYUSf97kDsZnx5wbdeUN5OtPzNAjKDTNt0kzGD5i1B
JCbUnUkY57n/aSbcqyML936cfpUYGIsOrElyj9LhizinQCVJXd4JUuA8y0WBSuvt
F2dn6ejoG80cy/G2vq4rA89DSKzWbaDwaX3hBsQcIeChOyb/Fa8+wBzkZoYj49Ov
Sk9U4eFhJ4mmFUzodogcH19X44daxlMm/rr2hPrTepE11MBrYmcJzI5oEiqrxGw7
ir3htP5wsF8BXepUV+/U6yKGoW6cXlMBLkmCQFyLF6sjqrfDaJpfLp41rtVJR7R5
b8W5/dKuvsFpJR7XYldMMN8nzTqI1mR/22uqlTukXTqLdX4RW7pLruFxRjkh41eN
KpETDUCB24UoEFEk3SmED28lAJshZ81SusDdWIkyZ2spnmwSHDYI2K0lzS/CcKbS
gwEJG8ocEjT7RMTzO2kOig8iVr33EXG8IWUQgwF0q7buYK2f+b+fQG8+MPPvK8gx
TqwIoNf6Jm7UdfgxjnVatUCsA5DcRaNXFY0O5PcCA4YKu6dqEBIig34Man3vEQXG
0G12a+5Qam3PCAG0zIuMC3NDOKbT8AQA/+1CjxNq2OqeLvVa
=6H5F
-----END PGP MESSAGE-----
```

**Comment by cielavenir** [comment]
to minimize the patch diff, I will continue Py3 with this branch enabled.

**Comment by ziyan** [comment]
@cielavenir Do we need any update to the rest of the system after this change?

**Comment by cielavenir** [comment]
> @cielavenir Do we need any update to the rest of the system after this change?

sorry for late reply, but for UI I don't think we need further updates. But I have not checked detectors/planning yet.

**Comment by rdiankov** [comment]
@cielavenir let me know when you tested. Thanks~

**Comment by cielavenir** [comment]
> But I have not checked detectors/planning yet.

I checked planningslave and so far this change looks reasonable. Will check detector after planningslave is merged.

**Comment by cielavenir** [comment]
I found another issue, so I change my branch name from now on.

File: CHANGELOG.md (hunk 1 above)
Comment (cielavenir) [reply CHANGELOG.md:3]: added minor, pushed

**Comment by cielavenir** [comment]
```
-----BEGIN PGP MESSAGE-----

hQIMAwAinXiLUbjzAQ/+PHN50cTAPLeKCVHm6rGvP6kj6LVyh9JCHfvvy5JtZ9w+
DGnxssqt6mjhp1BBNzZ4TctKM0AahyvZkB0ygg+fYorogZPxFev7c93HKThjDS1i
HdLr3B9Gbab1HxC1zDHTIRvNfdCHzUNalRAhcg1Q7P97B6h8RJEHc4hQiB5j74HV
V9DZAtpWHcuzbLHx62LT+AgyqTga1eDEu6Hkjuc04+uvnwz8AkJS6xwFdlv7zqDI
JPRPns2dqbyhIurpEn1A8L1qzJEP9ZMp45on31DptooPvRuRevw0Zr9KdW3/YQWw
+MSqX+AjqlCGU8agZvGkUFJJWhVOn6VIrfpKcfxLQXT/iwoM+fhki6b/mrs5i0vs
HNFfN/Rat/0sFPRHGiwFQC0mpAds4VyinU4eFPnVlo+YycklW5MvhE6D3kYZxF92
9Iwv52Wc82L6OMRKOhor3nNnzyQoNQPIWLHAFd0DhnDu6W7mudIKr2b8TEeYDLZt
DC0YmHpXwKg+QwUT3Evye5db2Y1zYJVOkxxb8X8Z+Kl0TNEpOV2balXSXc2YpFE8
Z23fAX/SCuJKuvBxG70KDpMSxyuqhbjU16oKCI+R1chdELaZVpGnGJPNgwn2cSVh
I0Y75fCB+2ZY+umHFN8x1U2pFwZsyYmQPvCs0n6gXZuVgJJFpRTRfX2gk9KEPDDS
ggHf4hWFFl5Fo2M2+akNSYENeXLMF742308lPJAFk5ctO3NMDoH17P0luOgY1djk
Rr7ZrvuYTYgIYh5oBseWh/R6aSBy7iKLDeDbtVYiydq92YNPQqVFPmPGNyFiop8i
t5osRnZ2/LN+2VM+mknRCObDUwLrLNCEXeZbij5AB80aFbc=
=7+gb
-----END PGP MESSAGE-----
```

**Comment by cielavenir** [comment]
I checked UI and planningslave and so far this change looks reasonable

**Comment by cielavenir** [comment]
/cc @ntohge

version.py started to conflict, I might need to revert the change...

----

He is now one of the leads, should be added to mujin group @rdiankov @liuhuanjim013

**Comment by cielavenir** [comment]
conflict started to happen, let me close it and move to https://github.com/mujin/mujinwebstackclientpy/pull/27 (I wanted to have 22 merged then 27)

**Pull request closed without merging**